### PR TITLE
plugins/lsp: add ruby-lsp

### LINKED ISSUE #1604: [PLUGIN REQUEST] ruby-lsp support
| Field    | Description |
|----------|-------------|
| Plugin   | `ruby-lsp`  | 
| Homepage | https://github.com/Shopify/ruby-lsp     | 
| Nixpkgs  | [`ruby-lsp`](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/top-level/ruby-packages.nix#L3294) | 

## Extra Information

Hi, I'm wondering if by any chance you can add support for ruby-lsp, it is a newer alternative to solargraph that is supposed to be more accurate. I'm very new to nixos and nixvim, so apologies if I'm missing details in the issue description.

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for [ruby-lsp](https://github.com/Shopify/ruby-lsp).

Fixes #1604

### CODE REVIEW

File: plugins/lsp/language-servers/default.nix [hunk 1]
```diff
@@ -502,6 +502,11 @@ let
       serverName = "rnix";
       package = null;
     }
+    {
+      name = "ruby-lsp";
+      description = "ruby-lsp for Ruby";
+      serverName = "ruby_lsp";
```
Comment (MattSturgeon) [inline plugins/lsp/language-servers/default.nix:508]: Should this define a package (`pkgs.ruby-lsp`) or is that handled automatically elsewhere?

**Review by MattSturgeon — APPROVED** [review]
I'm not confident  in this part of the codebase, but this looks consistent with outher definitions. If CI passes I'm happy.

File: plugins/lsp/language-servers/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/lsp/language-servers/default.nix:508]: This is done automatically yes: https://github.com/nix-community/nixvim/blob/0ba2ea5416485d3f6552a108710711acd688cfef/plugins/lsp/language-servers/_mk-lsp.nix#L12

**Comment by GaetanLepage** [comment]
> I'm not confident in this part of the codebase, but this looks consistent with outher definitions. If CI passes I'm happy.

Yes, we do this quite consistently in a straight forward way.

**Pull request merged**